src/chat.rs

```diff
@@ -22,13 +22,147 @@ use crate::param::*;
+
+impl ChatId {
+    /// Create a new [ChatId].
+    pub fn new(id: u32) -> ChatId {
+        ChatId(id)
+    }
+
+    /// A ChatID which indicates an error.
+    ///
+    /// This is transitional and should not be used in new code.  Do
+    /// not represent errors in a ChatId.
+    pub fn is_error(self) -> bool {
+        self.0 == 0
+    }
+
+    /// An unset ChatId
+    ///
+    /// Like [is_error], from which it is indistinguishable, this is
+    /// transitional and should not be used in new code.
+    pub fn is_unset(self) -> bool {
+        self.0 == 0
+    }
+
+    /// Whether the chat ID signifies a special chat.
+    ///
+    /// This kind of chat ID can not be used for real chats.
+    pub fn is_special(self) -> bool {
+        match self.0 {
+            0..=DC_CHAT_ID_LAST_SPECIAL => true,
+            _ => false,
+        }
+    }
+
+    /// Chat ID which represents the deaddrop chat.
+    ///
+    /// This is a virtual chat showing all messages belonging to chats
+    /// flagged with [Blocked::Deaddrop].  Usually the UI will show
+    /// these messages as contact requests.
+    pub fn is_deaddrop(self) -> bool {
+        self.0 == DC_CHAT_ID_DEADDROP
+    }
+
+    /// Chat ID for messages which need to be deleted.
+    ///
+    /// Messages which should be deleted get this chat ID and are
+    /// deleted later.  Deleted messages need to stay around as long
+    /// as they are not deleted on the server so that their rfc724_mid
+    /// remains known and downloading them again can be avoided.
+    pub fn is_trash(self) -> bool {
+        self.0 == DC_CHAT_ID_TRASH
+    }
+
+    // DC_CHAT_ID_MSGS_IN_CREATION seems unused?
+
+    /// Virtual chat showing all starred messages.
+    pub fn is_starred(self) -> bool {
+        self.0 == DC_CHAT_ID_STARRED
+    }
+
+    /// Chat ID signifying there are **any** number of archived chats.
+    ///
+    /// This chat ID can be returned in a [Chatlist] and signals to
+    /// the UI to include a link to the archived chats.
+    pub fn is_archived_link(self) -> bool {
+        self.0 == DC_CHAT_ID_ARCHIVED_LINK
+    }
+
+    /// Virtual chat ID signalling there are **only** archived chats.
+    ///
+    /// This can be included in the chatlist if the
+    /// [DC_GCL_ADD_ALLDONE_HINT] flag is used to build the
+    /// [Chatlist].
+    pub fn is_alldone_hint(self) -> bool {
+        self.0 == DC_CHAT_ID_ALLDONE_HINT
+    }
+
+    /// Bad evil escape hatch.
+    ///
+    /// Avoid using this, eventually types should be cleaned up enough
+    /// that it is no longer necessary.
+    pub fn to_u32(self) -> u32 {
+        self.0
+    }
+}
+
+impl std::fmt::Display for ChatId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.is_deaddrop() {
+            write!(f, "Chat#Deadrop")
+        } else if self.is_trash() {
+            write!(f, "Chat#Trash")
+        } else if self.is_starred() {
+            write!(f, "Chat#ArchivedLink")
```

Comment:
Starred on IMAP means it is archived? It is really not obvious and needs a comment, or maybe a mistake.

Reply:
i think this is just a mistake. it should read `if is_archived_link()  { write!(f, "Chat#ArchivedLink") } else if self.is_starred() { write!(f, "Chat#Starred") } ....

maybe this chat.is_starred() is also a bit misleading, not the chat is starred but the chat shows starred messages.

nb: starred messages are currently not used at all.